src/libraries/System.Data.Common/src/System/Data/SQLTypes/SqlXml.cs

```diff
@@ -424,6 +424,22 @@ public override int Read(byte[] buffer, int offset, int count)
             return iBytesRead;
         }
 
+        // Duplicate the Read(byte[]) logic here instead of refactoring both to use Spans
+        // in case the backing _stream doesn't override Read(Span).
```

Comment:
In most of our library code we've just assumed that any stream that cares about perf will end up overriding the relevant methods, and so for example wrapper streams like DeflateStream just use, say, `ReadAsync(Memory)` rather than having two code paths.  But if we think it's worthwhile here to special-case it, maybe because we expect streams wrapped here to be more legacy, that's ok.  Eventually it'd be nice to consolidate, though.

Reply:
Yeah, that was my thinking - this stream can come from a public constructor parameter, and this type is rather old - so older code could be calling it.

I took a quick look, and the first place I found actually has this problem:

https://github.com/dotnet/SqlClient/blob/9a7d53ea1651b94783c07f37eaa39cd8f728c02e/src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlCachedBuffer.cs#L130-L134

```C#
        internal SqlXml ToSqlXml()
        {
            SqlXml sx = new(ToStream());
            return sx;
        }
```

https://github.com/dotnet/SqlClient/blob/9a7d53ea1651b94783c07f37eaa39cd8f728c02e/src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlCachedBuffer.cs#L99-L102

```C#
        internal Stream ToStream()
        {
            return new SqlCachedStream(this);
        }
```

and [`SqlCachedStream`](https://github.com/dotnet/SqlClient/blob/9a7d53ea1651b94783c07f37eaa39cd8f728c02e/src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlStream.cs#L314) doesn't override the Span-based API.

I've opened https://github.com/dotnet/SqlClient/issues/1628 for the Sql team to consider providing Span-based overrides.